tests/unit/test-class-wc-payment-gateway-wcpay.php

```diff
@@ -944,6 +944,97 @@ public function test_capture_charge_without_level3() {
 		$this->assertEquals( Order_Status::PROCESSING, $order->get_status() );
 	}
 
+	public function test_capture_cancelling_order_cancels_authorization() {
+		$intent_id = uniqid( 'pi_' );
+		$charge_id = uniqid( 'ch_' );
+
+		$order = WC_Helper_Order::create_order();
+		$order->set_transaction_id( $intent_id );
+		$order->update_meta_data( '_intent_id', $intent_id );
+		$order->update_meta_data( '_charge_id', $charge_id );
+		$order->update_meta_data( '_intention_status', Intent_Status::REQUIRES_CAPTURE );
+		$order->update_status( Order_Status::ON_HOLD );
```

Comment:
Hm, are we sure about this status in this test case?

Reply:
That's the usual order status if payment hasn't been captured yet.